pandas/core/frame.py

```diff
@@ -5980,6 +5980,49 @@ def melt(self, id_vars=None, value_vars=None, var_name=None,
                     var_name=var_name, value_name=value_name,
                     col_level=col_level)
 
+    def explode(self, col_name, sep=None, dtype=None):
```

Comment:
Hmm would this be better as a Series method? Requiring `col_name` as a parameter makes it so it only operates as such anyway, no?

Reply:
Started reviewing before I saw your above comment. Still think this is better served as a Series method instead of a frame method with a required col_name argument. 

I think this would fail in cases where `col_name` is not unique

Reply:
I get your point though I think it would be better to simply return an object that a user can join themselves rather than try to take care of the merging within the method.

It's entirely reasonable to expect this against a Series object, so not offering that I think makes for a more confusing API.

Reply:
I agree it is an entirely plausible/reasonable scenario. However I would prefer to wait until we see people asking about `Series.explode` on github/mailing-list/stackoverflow to add that to Series. Otherwise if people only actually ever reach for `explode` in the context of a `DataFrame` then why bother having it in Series?

Note that this is also consistent with SQL / Spark APIs so I think it's unlikely for a lot of confusion to arise

Reply:
I agree this should be a Series method.